trieste/models/model_interfaces.py

```diff
@@ -274,13 +274,18 @@ def update(self, dataset: Dataset):
         assert dataset.observations.shape[-1] == y.shape[-1]
         data = (dataset.query_points, dataset.observations)
         num_data = data[0].shape[0]
-        num_latent_gps = model.num_latent_gps
+
+        new_q_mu, new_q_var = self.model.predict_f(dataset.query_points, full_cov=True)
```

Comment:
You probably want to be careful and assert that `self.model.q_sqrt.shape.ndims == 3` (i.e. `q_diag=False`) as GPflow also support only learning the marginals of `u` which results in a 2-dim `q_sqrt` (`self.model.q_sqrt.shape.ndims == 2`).

Reply:
Can you also assert that `white == False` for this model (not sure this applicable for VGP's though).

Reply:
I think that VGP does not have a whiten option

Reply:
sorry, actually VGP always uses white=True for conditionals

Reply:
When whitening (as @vpicheny pointed out hard-coded for VGP), predict_f and q_mu/q_sqrt live in different spaces. u = f(X) still lives in the same space, but with whitening we have u = L v, where L = cholesky(Kuu), and q(v) = N(q_mu, q_sqrt q_sqrt^T) (p(v) = N(0, I)). You get the correct q(u) = N(f_mean, f_cov) from f_mean, f_cov = predict_f(X, full_cov=True). They are related to q(v) by f_mean = L q_mu, f_cov = L q_sqrt q_sqrt^T L^T. To update q(v), you need to backtransform: q_mu = L^{-1} f_mean and q_sqrt = cholesky(S) where S = L^{-1} f_cov L^{-T}.
Alternatively, you could just use an SVGP with Z=X and whiten=False.

Reply:
Yes, what Ti is saying is exactly right! I've written some code before to do this, so copy-pasting it here for your help:
```python
# predict with old model
 f_mu, f_cov = self.model.predict_f(X_new, full_cov=True) # [N, L], [L, N, N]

Knn = model.kernel(X_new, full_cov=True)  # [N, N]
Lnn = tf.linalg.cholesky(Knn +  tf.eye(num_data, dtype=Knn.dtype) * jitter)  # [N, N]
q_mu_v = tf.linalg.triangular_solve(Lnn, f_mu)  # [N, L]
tmp = tf.linalg.triangular_solve(Lnn[None], f_var)  # [L, N, N], L^{-1} f_cov
S_v = tf.linalg.triangular_solve(Lnn[None], tf.linalg.matrix_transpose(tmp))  # [L, N, N]
new_q_sqrt = tf.linalg.cholesky(S_v + tf.eye(num_data, dtype=Knn.dtype) * jitter)  # [L, N, N]
```
Equations:
```
u = L * v, with v = N(0, 1) and K(X, X) = L L^T
q_mu_v = L^{-1} q_mu
q_sqrt_v = L^{-1} q_cov L^{-T}, q_cov = q_sqrt @ q_sqrt
```

Reply:
Great, you've sorted the jitter/dimensions aspect of it (just need to replace `f_var` -> `f_cov`). :)

Reply:
updated with your suggested code - thx a bunch